Client/Logging/ImpressionLoggerModifier.swift

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 Neeva Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+import Shared
+import SwiftUI
+
+struct ImpressionLoggerModifier: ViewModifier {
+    @State var impressionTimer: Timer? = nil
+
+    let timeInterval: TimeInterval
```

Comment:
We can default this interval to 2 seconds and let people override this interval if they want. That way, this should bring more consistent across app when no one overrides this :P

Reply:
done, i used the default in zero query view, but i kept my own for cheatsheet because the overlay thing is finicky.

Reply:
```suggestion
    let timeInterval = defaultTimeInterval
```

And this variable will be optional?

Reply:
but if i do that, i still need to keep the initializer around, else it wont compile properly if we pass a value to override it, so i feel that it suffices to put the default value as args to the initializer